views.go

```diff
@@ -117,23 +117,83 @@ func ProcessViewResult(result ViewResult, params map[string]interface{},
 	return result, nil
 }
 
-func ReduceViewResult(reduceFunction string, result *ViewResult) error {
+func ReduceViewResult(reduceFunction string, params map[string]interface{}, result *ViewResult) error {
+	reduceFun, compileErr := ReduceFunc(reduceFunction)
+	if compileErr != nil {
+		return compileErr
+	}
+	groupLevel := 0
+	if params["group"] != nil && params["group"].(bool) == true {
+		groupLevel = -1
+	} else if params["group_level"] != nil {
+		groupLevel = int(params["group_level"].(uint64))
+	}
+	if groupLevel != 0 {
+		var collator JSONCollator
+		key := result.Rows[0].Key
+		if groupLevel != -1 {
+			// don't try to cast key as a slice if group=true
+			key = key.([]interface{})[0:groupLevel]
```

Comment:
I want to take the prefix of the key, but only if the groupLevel parameter is greater than 0. I'm a little bit not sure exactly how I'd annotate this type for the compiler. Because I don't want to type assert unless the groupLevel is > 0